jbig2/jbig2_compress_image.go

```diff
@@ -75,8 +79,15 @@ func main() {
 		log.Fatalf("Error: %v\n", err)
 	}
 
+	fileNameWithoutExtension := func(filename string) string {
+		if i := strings.LastIndex(filename, "."); i != -1 {
+			return filename[:i]
+		}
+		return filename
+	}
+
 	// Write encoded data into a file with the extension '.jb2' - this is standard extension for the jbig2 files.
-	encodedFile, err := os.Create("checkerboard-squares-black-white.jb2")
+	encodedFile, err := os.Create(fileNameWithoutExtension(fileName) + ".jb2")
```

Comment:
I think filepath package has this included.
strings.TrimSuffix(filename, filepath.Ext(filename)) + ".jb2"

Reply:
Oh. That's nice solution, thanks. Done.